tests/core/test_device_save_loader.py

```diff
@@ -0,0 +1,197 @@
+from enum import Enum
+from os import path
+from typing import Any, Dict, List
+
+import numpy as np
+import numpy.typing as npt
+import pytest
+import yaml
+from bluesky.run_engine import RunEngine
+
+from ophyd_async.core import Device, SignalR, SignalRW
+from ophyd_async.core.device_save_loader import (
+    get_signal_values,
+    save_to_yaml,
+    walk_rw_signals,
+)
+from ophyd_async.epics.signal import epics_signal_r, epics_signal_rw
+
+
+class DummyChildDevice(Device):
+    def __init__(self) -> None:
+        self.sig1: SignalRW = epics_signal_rw(str, "Value1")
+        self.sig2: SignalR = epics_signal_r(str, "Value2")
+
+
+class EnumTest(str, Enum):
+    VAL1 = "val1"
+    VAL2 = "val2"
+
+
+class DummyDeviceGroup(Device):
+    def __init__(self, name: str):
+        self.child1: DummyChildDevice = DummyChildDevice()
+        self.child2: DummyChildDevice = DummyChildDevice()
+        self.parent_sig1: SignalRW = epics_signal_rw(str, "ParentValue1")
+        self.parent_sig2: SignalR = epics_signal_r(
+            int, "ParentValue2"
+        )  # Ensure only RW are found
+        self.parent_sig3: SignalRW = epics_signal_rw(str, "ParentValue3")
+        self.position: npt.NDArray[np.int32]
+
+
+@pytest.fixture
+async def device() -> DummyDeviceGroup:
+    device = DummyDeviceGroup("parent")
+    await device.connect(sim=True)
+    return device
+
+
+@pytest.fixture
+async def device_with_phases() -> DummyDeviceGroup:
+    device = DummyDeviceGroup("parent")
+    await device.connect(sim=True)
+
+    # Dummy function to check different phases save properly
+    def sort_signal_by_phase(self, values: Dict[str, Any]) -> List[Dict[str, Any]]:
+        phase_1 = {"child1.sig1": values["child1.sig1"]}
+        phase_2 = {"child2.sig1": values["child2.sig1"]}
+        return [phase_1, phase_2]
+
+    setattr(device, "sort_signal_by_phase", sort_signal_by_phase)
+    return device
+
+
+async def test_enum_yaml_formatting(tmp_path):
+    enums = [EnumTest(EnumTest.VAL1), EnumTest(EnumTest.VAL2)]
```

Comment:
I think it's easier to read this as:
```suggestion
    enums = [EnumTest.VAL1, EnumTest.VAL2]
```

Reply:
Ah you're right, was mistaken about how Python Enums work